src/analysis/von_neumann_algebra/basic.lean

```diff
@@ -0,0 +1,116 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import analysis.normed_space.dual
+import analysis.normed_space.star.basic
+import analysis.complex.basic
+import analysis.inner_product_space.adjoint
+
+/-!
+# Von Neumann algebras
+
+We give the "abstract" and "concrete" definitions of a von Neumann algebra.
+We still have a major project ahead of us to show the equivalence between these definitions!
+
+An abstract von Neumann algebra `wstar_algebra M` is a C^* algebra with a Banach space predual,
+per Sakai (1971).
+
+A concrete von Neumann algebra `von_neumann_algebra H` (where `H` is a Hilbert space)
+is a *-closed subalgebra of bounded operators on `H` which is equal to its double commutant.
+
+We'll also need to prove the von Neumann double commutant theorem,
+that the concrete definition is equivalent to a *-closed subalgebra which is weakly closed.
+-/
+
+universes u v
+
+/--
+Sakai's definition of a von Neumann algebra as a C^* algebra with a Banach space predual.
+
+So that we can unambiguously talk about these "abstract" von Neumann algebras
+in parallel with the "concrete" ones (weakly closed *-subalgebras of B(H)),
+we name this definition `wstar_algebra`.
+-/
+class wstar_algebra (M : Type u) [normed_ring M] [star_ring M] [cstar_ring M]
+  [module ℂ M] [normed_algebra ℂ M] [star_module ℂ M] :=
+(exists_predual : ∃ (X : Type u) [normed_group X] [normed_space ℂ X] [complete_space X],
+  nonempty (normed_space.dual ℂ X ≃ₗᵢ⋆[ℂ] M))
```

Comment:
I completely understand not wanting the predual to be data, because then `wstar_algebra`s would need to have the same predual in order to be equal. However, I think it might actually be the right approach. Reasoning: we'll need to access the weak star topology on the W-star algebra induced by the predual. If the predual isn't data, we'll have to use `classical.choice` (or analogous) to get a predual on which we can instantiate the weak star topology. So, should `wstar_algebra` actually carry around the predual as data, or do we use `classical.choice` to get a non-canonical (but fixed) predual from which we get the weak star topology?

Reply:
Interesting. I guess this comes down to whether we care about definitional properties of the weak-* star topology.

I've never encountered needing this in "real maths", but I also haven't thought much about W-star algebras per se... I've always happily lived in the concrete setting. :-)

Reply:
I've added an extensive comment explaining this concern. My inclination for now to to hope that we can cope with `classical.choice` appearing inside the definition of the weak-* topology, and revisit this if we run into trouble.